docs/user-guide/syntheticdata.rst

```diff
@@ -14,5 +14,743 @@ There are a variety of ways to construct synthetic data generation pipelines, wi
 
 NeMo Curator has a simple, easy-to-use set of tools that allow you to use prebuilt synthetic generation pipelines or build your own.
 Any model inference service that uses the OpenAI API is compatible with the synthetic data generation module, allowing you to generate your data from any model.
+Furthermore, NeMo Curator also can interface with `NeMo's Export and Deploy <https://docs.nvidia.com/nemo-framework/user-guide/latest/deployingthenemoframeworkmodel.html#use-nemo-export-and-deploy-module-apis-to-run-inference>`_
+module which allows you to host your own model for LLM inference.
 NeMo Curator has prebuilt synthetic data generation pipelines for supervised fine-tuning (SFT) and preference data that were used to generate data for the training of `Nemotron-4 340B <https://research.nvidia.com/publication/2024-06_nemotron-4-340b>`_.
-And, you can easily interweave filtering and deduplication steps in your synthetic data pipeline with the other modules in NeMo Curator.
\ No newline at end of file
+And, you can easily interweave filtering and deduplication steps in your synthetic data pipeline with the other modules in NeMo Curator.
+
+--------------------------------------
+Connecting to an LLM Service
+--------------------------------------
+NeMo Curator supports connecting to `OpenAI API <https://github.com/openai/openai-python?tab=readme-ov-file#openai-python-api-library>`_ compatible services and `NeMo Deploy <https://docs.nvidia.com/nemo-framework/user-guide/latest/deployingthenemoframeworkmodel.html#use-nemo-export-and-deploy-module-apis-to-run-inference>`_ services.
+Despite its name, the OpenAI API is used for querying models across different platforms beyond OpenAI's own models.
+Here is how we can connect to `build.nvidia.com <https://build.nvidia.com/explore/discover>`_ to query Gemma 2 9b-it using NeMo Curator and the OpenAI API.
+
+.. code-block:: python
+
+    from openai import OpenAI
+    from nemo_curator import OpenAIClient
+
+    openai_client = OpenAI(
+        base_url="https://integrate.api.nvidia.com/v1",
+        api_key="<insert API key>",
```

Comment:
```suggestion
        api_key="<insert OpenAI API key>",
```

Reply:
I don't want people to get confused that the API key comes from OpenAI, but I do see that we need to clarify it. I've changed it to `<insert NVIDIA API key>` instead.